server/src/services/library.service.ts

```diff
@@ -366,258 +358,208 @@ export class LibraryService {
     return JobStatus.SUCCESS;
   }
 
-  async handleAssetRefresh(job: ILibraryFileJob): Promise<JobStatus> {
-    const assetPath = path.normalize(job.assetPath);
-
-    const existingAssetEntity = await this.assetRepository.getByLibraryIdAndOriginalPath(job.id, assetPath);
-
-    let stats: Stats;
+  private async getMtime(path: string): Promise<Date> {
     try {
-      stats = await this.storageRepository.stat(assetPath);
+      const stat = await this.storageRepository.stat(path);
+      return stat.mtime;
     } catch (error: Error | any) {
-      // Can't access file, probably offline
-      if (existingAssetEntity) {
-        // Mark asset as offline
-        this.logger.debug(`Marking asset as offline: ${assetPath}`);
-
-        await this.assetRepository.update({ id: existingAssetEntity.id, isOffline: true });
-        return JobStatus.SUCCESS;
-      } else {
-        // File can't be accessed and does not already exist in db
-        throw new BadRequestException('Cannot access file', { cause: error });
-      }
+      throw new BadRequestException(`Cannot access file ${path}`, { cause: error });
     }
+  }
 
-    let doImport = false;
-    let doRefresh = false;
+  async handleAssetRefresh(job: ILibraryFileJob): Promise<JobStatus> {
+    const assetPath = path.normalize(job.assetPath);
 
-    if (job.force) {
-      doRefresh = true;
-    }
+    let asset = await this.assetRepository.getByLibraryIdAndOriginalPath(job.id, assetPath);
 
     const originalFileName = parse(assetPath).base;
 
-    if (!existingAssetEntity) {
-      // This asset is new to us, read it from disk
-      this.logger.debug(`Importing new asset: ${assetPath}`);
-      doImport = true;
-    } else if (stats.mtime.toISOString() !== existingAssetEntity.fileModifiedAt.toISOString()) {
-      // File modification time has changed since last time we checked, re-read from disk
-      this.logger.debug(
-        `File modification time has changed, re-importing asset: ${assetPath}. Old mtime: ${existingAssetEntity.fileModifiedAt}. New mtime: ${stats.mtime}`,
-      );
-      doRefresh = true;
-    } else if (existingAssetEntity.originalFileName !== originalFileName) {
-      // TODO: We can likely remove this check in the second half of 2024 when all assets have likely been re-imported by all users
-      this.logger.debug(
-        `Asset is missing file extension, re-importing: ${assetPath}. Current incorrect filename: ${existingAssetEntity.originalFileName}.`,
-      );
-      doRefresh = true;
-    } else if (!job.force && stats && !existingAssetEntity.isOffline) {
-      // Asset exists on disk and in db and mtime has not changed. Also, we are not forcing refresn. Therefore, do nothing
-      this.logger.debug(`Asset already exists in database and on disk, will not import: ${assetPath}`);
-    }
+    if (asset) {
```

Comment:
I still find this quite hard to read/complex overall (lots of nesting/conditions).